pkg/xen-tools/xen-start

```diff
@@ -31,8 +31,15 @@ echo running > "/run/tasks/$1"
 # FIXME: this really needs to be managed by runc
 keyctl link @u @s
 
+# if HVM capabilities exist, we run PV guests as PVH
+if grep -q 'type.*=.*"pv"' "$2"; then
+   # shellcheck disable=SC2089
+   xl info | grep -q 'virt_caps.*hvm' && EXTRA_ARGS='type="pvh"'
+fi
+
 # create domain in a paused state
-xl create "$2" -p || bail "xl created failed"
+# shellcheck disable=SC2090
+xl create "$2" -p $EXTRA_ARGS || bail "xl created failed"
```

Comment:
Doesn't this mean the command line will have type=pv -p type=pvh?
Seems a bit odd but fine if it works.

Reply:
Yes -- that's the idea -- to basically provide an override based on discovered capabilities @eriknordmark